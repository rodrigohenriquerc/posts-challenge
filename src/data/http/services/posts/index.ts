export * from './getPosts'
export * from './getPosts/protocol'
