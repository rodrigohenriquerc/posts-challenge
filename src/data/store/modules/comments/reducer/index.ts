import { ActionTypes, CommentsState, Action } from '../types'

export const INITIAL_STATE: CommentsState = {
  data: [],
  error: null,
  isLoading: true,
}

const CommentsReducer = (state = INITIAL_STATE, action: Action) => {
  switch (action.type) {
    case ActionTypes.GET_COMMENTS_REQUEST:
      return {
        ...state,
        error: null,
        isLoading: true,
      }
    case ActionTypes.GET_COMMENTS_SUCCESS:
      return {
        ...state,
        data: action.payload.data,
        error: null,
        isLoading: false,
      }
    case ActionTypes.GET_COMMENTS_FAILURE:
      return {
        ...state,
        error: action.payload.error,
        isLoading: false,
      }
    default:
      return state
  }
}

export default CommentsReducer
