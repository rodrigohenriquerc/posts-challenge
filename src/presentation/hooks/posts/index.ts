export { default as usePostsSelector } from './usePostsSelector'
export { default as usePostsDispatch } from './usePostsDispatch'
