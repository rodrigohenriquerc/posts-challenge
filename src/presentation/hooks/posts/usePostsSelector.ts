import { useStateSelector } from '../state/useStateSelector'

export default function usePostsSelector() {
  const { data, isLoading } = useStateSelector((state) => state.posts)
  return { data, isLoading }
}
