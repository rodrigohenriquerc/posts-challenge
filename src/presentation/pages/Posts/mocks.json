{
  "posts": [
    {
      "author": "Rodrigo Henrique",
      "avatar": "https://media-exp1.licdn.com/dms/image/C4D03AQHfEYzGBpTUxQ/profile-displayphoto-shrink_800_800/0/1607482823086?e=1620864000&v=beta&t=zBaA3SbRac1BV0f_nJn-5o7c5I8ERlr6PT19nFmQeuM",
      "title": "Como ganhar dinheiro",
      "description": "Para ganhar dinheiro, você deve estar disposto a abrir mão de muitas coisas. Você deve acordar cedo todos os dias e trabalhar muito.",
      "image": "https://animals.sandiegozoo.org/sites/default/files/2016-11/animals_hero_lizards.jpg"
    },
    {
      "author": "Rodrigo Henrique",
      "avatar": "https://media-exp1.licdn.com/dms/image/C4D03AQHfEYzGBpTUxQ/profile-displayphoto-shrink_800_800/0/1607482823086?e=1620864000&v=beta&t=zBaA3SbRac1BV0f_nJn-5o7c5I8ERlr6PT19nFmQeuM",
      "title": "Como ganhar dinheiro",
      "description": "Para ganhar dinheiro, você deve estar disposto a abrir mão de muitas coisas. Você deve acordar cedo todos os dias e trabalhar muito.",
      "image": "https://animals.sandiegozoo.org/sites/default/files/2016-11/animals_hero_lizards.jpg"
    },
    {
      "author": "Rodrigo Henrique",
      "avatar": "https://media-exp1.licdn.com/dms/image/C4D03AQHfEYzGBpTUxQ/profile-displayphoto-shrink_800_800/0/1607482823086?e=1620864000&v=beta&t=zBaA3SbRac1BV0f_nJn-5o7c5I8ERlr6PT19nFmQeuM",
      "title": "Como ganhar dinheiro",
      "description": "Para ganhar dinheiro, você deve estar disposto a abrir mão de muitas coisas. Você deve acordar cedo todos os dias e trabalhar muito.",
      "image": "https://animals.sandiegozoo.org/sites/default/files/2016-11/animals_hero_lizards.jpg"
    },
    {
      "author": "Rodrigo Henrique",
      "avatar": "https://media-exp1.licdn.com/dms/image/C4D03AQHfEYzGBpTUxQ/profile-displayphoto-shrink_800_800/0/1607482823086?e=1620864000&v=beta&t=zBaA3SbRac1BV0f_nJn-5o7c5I8ERlr6PT19nFmQeuM",
      "title": "Como ganhar dinheiro",
      "description": "Para ganhar dinheiro, você deve estar disposto a abrir mão de muitas coisas. Você deve acordar cedo todos os dias e trabalhar muito.",
      "image": "https://animals.sandiegozoo.org/sites/default/files/2016-11/animals_hero_lizards.jpg"
    },
    {
      "author": "Rodrigo Henrique",
      "avatar": "https://media-exp1.licdn.com/dms/image/C4D03AQHfEYzGBpTUxQ/profile-displayphoto-shrink_800_800/0/1607482823086?e=1620864000&v=beta&t=zBaA3SbRac1BV0f_nJn-5o7c5I8ERlr6PT19nFmQeuM",
      "title": "Como ganhar dinheiro",
      "description": "Para ganhar dinheiro, você deve estar disposto a abrir mão de muitas coisas. Você deve acordar cedo todos os dias e trabalhar muito.",
      "image": "https://animals.sandiegozoo.org/sites/default/files/2016-11/animals_hero_lizards.jpg"
    },
    {
      "author": "Rodrigo Henrique",
      "avatar": "https://media-exp1.licdn.com/dms/image/C4D03AQHfEYzGBpTUxQ/profile-displayphoto-shrink_800_800/0/1607482823086?e=1620864000&v=beta&t=zBaA3SbRac1BV0f_nJn-5o7c5I8ERlr6PT19nFmQeuM",
      "title": "Como ganhar dinheiro",
      "description": "Para ganhar dinheiro, você deve estar disposto a abrir mão de muitas coisas. Você deve acordar cedo todos os dias e trabalhar muito.",
      "image": "https://animals.sandiegozoo.org/sites/default/files/2016-11/animals_hero_lizards.jpg"
    },
    {
      "author": "Rodrigo Henrique",
      "avatar": "https://media-exp1.licdn.com/dms/image/C4D03AQHfEYzGBpTUxQ/profile-displayphoto-shrink_800_800/0/1607482823086?e=1620864000&v=beta&t=zBaA3SbRac1BV0f_nJn-5o7c5I8ERlr6PT19nFmQeuM",
      "title": "Como ganhar dinheiro",
      "description": "Para ganhar dinheiro, você deve estar disposto a abrir mão de muitas coisas. Você deve acordar cedo todos os dias e trabalhar muito.",
      "image": "https://animals.sandiegozoo.org/sites/default/files/2016-11/animals_hero_lizards.jpg"
    },
    {
      "author": "Rodrigo Henrique",
      "avatar": "https://media-exp1.licdn.com/dms/image/C4D03AQHfEYzGBpTUxQ/profile-displayphoto-shrink_800_800/0/1607482823086?e=1620864000&v=beta&t=zBaA3SbRac1BV0f_nJn-5o7c5I8ERlr6PT19nFmQeuM",
      "title": "Como ganhar dinheiro",
      "description": "Para ganhar dinheiro, você deve estar disposto a abrir mão de muitas coisas. Você deve acordar cedo todos os dias e trabalhar muito.",
      "image": "https://animals.sandiegozoo.org/sites/default/files/2016-11/animals_hero_lizards.jpg"
    },
    {
      "author": "Rodrigo Henrique",
      "avatar": "https://media-exp1.licdn.com/dms/image/C4D03AQHfEYzGBpTUxQ/profile-displayphoto-shrink_800_800/0/1607482823086?e=1620864000&v=beta&t=zBaA3SbRac1BV0f_nJn-5o7c5I8ERlr6PT19nFmQeuM",
      "title": "Como ganhar dinheiro",
      "description": "Para ganhar dinheiro, você deve estar disposto a abrir mão de muitas coisas. Você deve acordar cedo todos os dias e trabalhar muito.",
      "image": "https://animals.sandiegozoo.org/sites/default/files/2016-11/animals_hero_lizards.jpg"
    },
    {
      "author": "Rodrigo Henrique",
      "avatar": "https://media-exp1.licdn.com/dms/image/C4D03AQHfEYzGBpTUxQ/profile-displayphoto-shrink_800_800/0/1607482823086?e=1620864000&v=beta&t=zBaA3SbRac1BV0f_nJn-5o7c5I8ERlr6PT19nFmQeuM",
      "title": "Como ganhar dinheiro",
      "description": "Para ganhar dinheiro, você deve estar disposto a abrir mão de muitas coisas. Você deve acordar cedo todos os dias e trabalhar muito.",
      "image": "https://animals.sandiegozoo.org/sites/default/files/2016-11/animals_hero_lizards.jpg"
    }
  ]
}
