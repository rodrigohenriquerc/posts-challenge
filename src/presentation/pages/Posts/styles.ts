import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 2rem;
  background-color: #0D122B;
`;

export const List = styled.ul`
  max-width: 768px;
`;

export const ListItem = styled.li`
  list-style: none;
  margin-bottom: 2rem;
`;
