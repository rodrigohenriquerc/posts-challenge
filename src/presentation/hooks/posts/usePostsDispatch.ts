import { useDispatch } from 'react-redux'
import { GetPostsRequest, RemovePostRequest } from 'data/store/modules/posts/actions'

export default function usePostsDispatch() {
  const dispatch = useDispatch()

  const getPosts = () => {
    dispatch(GetPostsRequest())
  }

  const removePost = (postId: number) => {
    dispatch(RemovePostRequest({ postId }))
  }

  return { getPosts, removePost }
}
