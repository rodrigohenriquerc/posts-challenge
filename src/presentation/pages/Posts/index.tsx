import React, { useEffect } from 'react'
import { PostCard, Loading } from 'presentation/components'
import { IPostCard } from 'presentation/components/PostCard/models'
import * as PostsHooks from 'presentation/hooks/posts'
import * as S from './styles'

const Posts: React.FC = () => {
  const { getPosts } = PostsHooks.usePostsDispatch()
  const { data, isLoading } = PostsHooks.usePostsSelector()

  useEffect(() => getPosts(), [])

  return (
    <S.Container isLoading={isLoading}>
      {isLoading ? (
        <Loading />
      ) : (
        <S.List>
          {data.map((post: IPostCard) => (
            <S.ListItem key={post.id}>
              <PostCard
                id={post.id}
                author={post.author}
                title={post.title}
                description={post.description}
              />
            </S.ListItem>
          ))}
        </S.List>
      )}
    </S.Container>
  )
}

export default Posts
