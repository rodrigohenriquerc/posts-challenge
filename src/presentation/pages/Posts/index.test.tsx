import React from 'react'
import { storeFactory } from 'setupTests'
import { renderWithRedux } from 'utils/testConfig'
import { INITIAL_STATE } from 'data/store/modules/posts/reducer'
import { PostsState } from 'data/store/modules/posts/types'
import Posts from '.'

interface SystemUnderTest {
  mockStore: any
  createWrapper: any
}

const sut: SystemUnderTest = {
  mockStore: {},
  createWrapper: {},
}

const mockStore = (postsState?: PostsState) =>
  storeFactory({
    posts: postsState || INITIAL_STATE,
  })

describe('Posts', () => {
  beforeEach(() => {
    sut.mockStore = mockStore
    sut.createWrapper = (store: any) => renderWithRedux(<Posts />, { store })
  })
  it('should render properly', () => {
    const store = sut.mockStore()
    const wrapper = sut.createWrapper(store)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render loading UI when data is loading', () => {
    const store = sut.mockStore()
    const wrapper = sut.createWrapper(store)
    const loadingUI = wrapper.queryByTestId('loading-ui')
    expect(loadingUI).toBeTruthy()
  })
})
