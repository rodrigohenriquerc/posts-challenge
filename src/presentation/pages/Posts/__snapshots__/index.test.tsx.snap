// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts should render properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  display: inline-block;
  position: relative;
  width: 80px;
  height: 80px;
}

.c1 div {
  position: absolute;
  width: 1rem;
  height: 1rem;
  border-radius: 50%;
  background: #fff;
  -webkit-animation-timing-function: cubic-bezier(0,1,1,0);
  animation-timing-function: cubic-bezier(0,1,1,0);
}

.c1 div:nth-child(1) {
  left: 8px;
  -webkit-animation: lds-ellipsis1 0.6s infinite;
  animation: lds-ellipsis1 0.6s infinite;
}

.c1 div:nth-child(2) {
  left: 8px;
  -webkit-animation: lds-ellipsis2 0.6s infinite;
  animation: lds-ellipsis2 0.6s infinite;
}

.c1 div:nth-child(3) {
  left: 32px;
  -webkit-animation: lds-ellipsis2 0.6s infinite;
  animation: lds-ellipsis2 0.6s infinite;
}

.c1 div:nth-child(4) {
  left: 56px;
  -webkit-animation: lds-ellipsis3 0.6s infinite;
  animation: lds-ellipsis3 0.6s infinite;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-height: 100vh;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: border-box;
  padding: 1rem;
  padding-bottom: 0;
  background-color: #0d122b;
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
          data-testid="loading-ui"
        >
          <div />
          <div />
          <div />
          <div />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-iBPRYJ beOIYt"
    >
      <div
        class="sc-bdfBwQ isxrlz"
        data-testid="loading-ui"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
